### PR TITLE
Merge Network release branch into master

### PR DESCRIPTION
**Pull request by anton-evseev**
Required two other PRs to get merged first:
- https://github.com/Azure/azure-rest-api-specs/pull/5087
- https://github.com/Azure/azure-rest-api-specs/pull/5143

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by number213** [comment]
@MikhailTryakhov please resolved merge conflicts, I don't have write access

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/vmssNetworkInterface.json [hunk 1]
```diff
@@ -0,0 +1,389 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "NetworkManagementClient",
+    "description": "The Microsoft Azure Network management API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
+    "version": "2017-03-30"
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/vmssNetworkInterface.json]: 2018-12-01

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json [hunk 2]
```diff
@@ -0,0 +1,3348 @@
...
+          },
+          {
+            "name": "gatewayName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the gateway."
+          },
+          {
+            "$
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json]: provisisoning -> provisioning

Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json]: double word "for"

Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json]: vpnclients -> VPN clients'

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json [hunk 3]
```diff
@@ -0,0 +1,3348 @@
...
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/vpnSites/{vpnSiteName}": {
+      "get": {
+        "operationId": "VpnSites_Get",
+        "x-ms-examples": {
+          "VpnSiteGet": {
+            "$ref": "./examples/VpnSiteGet.json"
+          }
+        },
+        "description": "Retrieves the details of a VPNsite.",
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualWan.json]: VPNsite -> VPN site

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualNetworkGateway.json [hunk 4]
```diff
@@ -0,0 +1,2817 @@
...
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          },
+          "200": {
+            "description": "Request successful. The operation sets the specificed vpnclient ipsec parameters for P2S client of the virtual network gateway.",
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/virtualNetworkGateway.json]: specificed -> specified

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/serviceEndpointPolicy.json [hunk 5]
```diff
@@ -0,0 +1,647 @@
...
+          },
+          "200": {
+            "description": "Delete successful."
+          },
+          "204": {
+            "description": "Request successful. Resource does not exist."
+          }
+        },
+        "x-ms-examples": {
+          "Delete service endpoint Policys": { "$ref": "./examples/ServiceEndpointPolicyDelete.json" }
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/serviceEndpointPolicy.json]: Policys -> Policy

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/publicIpPrefix.json [hunk 6]
```diff
@@ -0,0 +1,440 @@
...
+              "required": true,
+              "type": "string",
+              "description": "The name of the resource group."
+            },
+            {
+              "name": "publicIpPrefixName",
+              "in": "path",
+              "required": true,
+              "type": "string",
+              "description": "The name of the PublicIPPrefx."
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/publicIpPrefix.json]: PublicIPPrefx -> Public IP Prefix

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkWatcher.json [hunk 7]
```diff
@@ -0,0 +1,3648 @@
...
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns the result of network condifuration diagnostic.",
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkWatcher.json]: condifuration -> configuration

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkWatcher.json [hunk 8]
```diff
@@ -0,0 +1,3648 @@
...
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ],
+      "description": "Network watcher in a resource group."
+    },
+    "NetworkWatcherPropertiesFormat": {
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkWatcher.json]: snaphots -> snapshots

Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkWatcher.json]: Traffice -> Traffic

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkSecurityGroup.json [hunk 9]
```diff
@@ -0,0 +1,848 @@
...
+          }
+        },
+        "priority": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The priority of the rule. The value can be between 100 and 4096. The priority number must be unique for each rule in the collection. The lower the priority number, the higher the priority of the rule."
+        },
+        "direction": {
+          "type": "string",
+          "description": "The direction of the rule. The direction specifies if rule will be evaluated on incoming or outcoming traffic. Possible values are: 'Inbound' and 'Outbound'.",
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkSecurityGroup.json]: outcoming -> outgoing

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json [hunk 10]
```diff
@@ -0,0 +1,575 @@
...
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "networkProfileName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the PublicIPPrefx."
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json]: PublicIPPrefx -> Public IP Prefix

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json [hunk 11]
```diff
@@ -0,0 +1,575 @@
...
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of NetworkProfile resources.",
+            "schema": {
+              "$ref": "#/definitions/NetworkProfileListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List all network profilees": {
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json]: profilees -> profiles

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json [hunk 12]
```diff
@@ -0,0 +1,575 @@
...
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of NetworkProfile resources.",
+            "schema": {
+              "$ref": "#/definitions/NetworkProfileListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List resource group network profilees": {
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json]: profilees -> profiles

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json [hunk 13]
```diff
@@ -0,0 +1,575 @@
...
+    },
+    "ContainerNetworkInterfacePropertiesFormat": {
+      "properties": {
+        "containerNetworkInterfaceConfiguration": {
+          "$ref": "#/definitions/ContainerNetworkInterfaceConfiguration",
+          "description": "Container network interface configuration from which this container network interface is created."
+        },
+        "container": {
+          "$ref": "#/definitions/Container",
+          "description": "Reference to the conatinaer to which this container network interface is attached."
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json]: conatinaer -> container

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json [hunk 14]
```diff
@@ -0,0 +1,575 @@
...
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "Container network interface configruation child resource."
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json]: configruation -> configuration

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json [hunk 15]
```diff
@@ -0,0 +1,575 @@
...
+    },
+    "IPConfigurationProfilePropertiesFormat": {
+      "properties": {
+        "subnet": {
+          "$ref": "./virtualNetwork.json#/definitions/Subnet",
+          "description": "The reference of the subnet resource to create a contatainer network interface ip configruation."
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json]: configruation -> configuration

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json [hunk 16]
```diff
@@ -0,0 +1,575 @@
...
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The provisioning state of the resource."
+        }
+      },
+      "description": "IP configruation profile properties."
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json]: configruation -> configuration

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json (hunk 15 above)
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkProfile.json]: contatainer -> container

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkInterface.json [hunk 17]
```diff
@@ -0,0 +1,1393 @@
...
+        "destinationPortRange": {
+          "type": "string",
+          "description": "The destination port or range."
+        },
+        "sourcePortRanges": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The source port ranges. Expected values include a single integer between 0 and 65535, a range using '-' as seperator (e.g. 100-400), or an asterix (*)"
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkInterface.json]: seperator -> separator

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkInterface.json [hunk 18]
```diff
@@ -0,0 +1,1393 @@
...
+        },
+        "destinationPortRanges": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The destination port ranges. Expected values include a single integer between 0 and 65535, a range using '-' as seperator (e.g. 100-400), or an asterix (*)"
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkInterface.json]: seperator -> separator

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkInterface.json [hunk 19]
```diff
@@ -0,0 +1,1393 @@
...
+        "destinationAddressPrefix": {
+          "type": "string",
+          "description": "The destination address prefix."
+        },
+        "sourceAddressPrefixes" : {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The source address prefixes. Expected values include CIDR IP ranges, Default Tags (VirtualNetwork, AureLoadBalancer, Internet), System Tags, and the asterix (*)."
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkInterface.json]: AureLoadBalancer -> AzureLoadBalancer

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkInterface.json [hunk 20]
```diff
@@ -0,0 +1,1393 @@
...
+        },
+        "destinationAddressPrefixes" : {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The destination address prefixes. Expected values include CIDR IP ranges, Default Tags (VirtualNetwork, AureLoadBalancer, Internet), System Tags, and the asterix (*)."
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkInterface.json]: AureLoadBalancer -> AzureLoadBalancer

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCrossConnection.json [hunk 21]
```diff
@@ -0,0 +1,922 @@
...
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of ExpressRouteCrossConnection resources. If there are no crossconnection resources an empty list is returned.",
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCrossConnection.json]: crossconnection -> cross connection

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCrossConnection.json [hunk 22]
```diff
@@ -0,0 +1,922 @@
...
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful.The operation returns a list of ExpressRouteCrossConnection resources. If there are no crossconnection resources an empty list is returned.",
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCrossConnection.json]: crossconnection -> cross connection

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCircuit.json [hunk 23]
```diff
@@ -0,0 +1,2019 @@
...
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of ExpressRouteServiceProdiver resources.",
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCircuit.json]: ExpressRouteServiceProdiver -> ExpressRouteServiceProvider

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCircuit.json [hunk 24]
```diff
@@ -0,0 +1,2019 @@
...
+            "type": "string"
+          },
+          "description": "The reference of AdvertisedPublicPrefixes."
+        },
+        "advertisedCommunities": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The communities of bgp peering. Spepcified for microsoft peering"
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCircuit.json]: Spepcified -> Specified

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCircuit.json [hunk 25]
```diff
@@ -0,0 +1,2019 @@
...
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "circuitConnectionStatus",
+            "modelAsString": true
+          }
+        },
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Provisioning state of the circuit connection resource. Possible values are: 'Succeded', 'Updating', 'Deleting', and 'Failed'."
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCircuit.json]: Succeded -> Succeeded as long as the typo isn't in the enum

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/vmssNetworkInterface.json (hunk 1 above)
Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/vmssNetworkInterface.json]: VMSS has it's own API version, this is expected

Comment (nschonni) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/vmssNetworkInterface.json]: Should it be under it's own version folder then?

Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/vmssNetworkInterface.json]: I don't know all the details, there's an open issue for that: https://github.com/Azure/azure-rest-api-specs/issues/1685

File: specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCircuit.json (hunk 25 above)
Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/expressRouteCircuit.json]: SDK code seems to override this prop with the one inherited from SubResource
https://github.com/Azure/azure-sdk-for-net/blob/master/src/SDKs/Network/Management.Network/Generated/Models/ExpressRouteCircuitConnection.cs#L52
I'll double check what the server returns

**Comment by number213** [comment]
Closing in favor of https://github.com/Azure/azure-rest-api-specs/pull/5189

**Pull request closed without merging**